### DISCUSSION TITLE
Multimonitor and workspace questions

### DISCUSSION THREAD

**Discussion by vlada-dudr** [hyprwm/Hyprland#835] [Q&A] [answered]
Hi all! Started to play with hyprland and I am trying to find my way around it. I have some questions, so I hope someone will help me out with those:
1) I managed to switch to arbitrary workspace on arbitrary monitor, but I'd love to exchange the workspaces if I am switching to workspace currently open on other output.
2) i'd love to use `movefocus` dispatcher to focus also empty monitors so I can just move there and open something there, if not possible, I could get around be "focus next monitor" command, but I didn't find such
3) is there way to have all workspaces created right from the beginning, so they show up in bar?
4) can Waybar (and i guess i should ask Waybar guys) show not "globally" active workspace, but workspace which is currently focused on respective output it is itself showing on?
5) Is there way to focus next workspace, ignoring output it belongs to?
6) obviously the thing I want is completely ignore the association of workspace to monitor (except where it is currently shown, of course)

**Comment by vaxerski** [comment, marked as answer]
1. discussed before:
```
bind=mod,key,moveworkspacetomonitor,1 current
bind=same mod, same key,workspace,1
```

2. u cant atm

3. no, empty workspaces not in focus get closed

4. no

5. yes. +1

**Reply by vlada-dudr** [reply]
> 1. discussed before:

That's what I used, but the swapping part?

Thank you for fast answer!

**Reply by vaxerski** [reply]
use a script to parse hyprctl workspaces and deal with that

**Reply by vlada-dudr** [reply]
Ok. Are you interested to include resulting script to contrib/wiki?

**Reply by vaxerski** [reply]
I'm sure @fufexan wouldn't have anything against that (he maintains contrib)

**Reply by Schievel1** [reply]
I wrote that script you wanted, see https://github.com/hyprwm/contrib/pull/32
